ext/standard/var.c

```diff
@@ -829,12 +835,6 @@ static int php_var_serialize_get_sleep_props(
 			continue;
 		}
 
-		if (EG(exception)) {
```

Comment:
These shouldn't be removed? In php, a lot of operations can throw exceptions, so EG(exception) usually has a reason.

Here, if php_var_serialize_try_add_sleep_prop emits an E_NOTICE for duplicate properties, and set_error_handler is a function that always throws an exception/error, then EG(exception) would be set for `return ['prop', 'prop'];`

Reply:
These blocks weren't here before https://github.com/php/php-src/commit/846b6479537a112d1ded725e6484e46462048b35 and these exceptions were still handled fine, isn't it? Since the code to throw inside `php_var_serialize_get_sleep_props()` has been removed, what's the point of keeping these blocks?

Reply:
Will defer to @nikic on this.

First:

Prior to this PR, both C and PHP (from E_NOTICE) could throw.
After this PR, only PHP (E_NOTICE) could throw, but it was still possible.

Second:
I think that adding a sleep property should still throw when returning a non-nullable typed property that got unset (see other comment)

Reply:
> I think that adding a sleep property should still throw when returning a non-nullable typed property that got unset (see other comment)

That's a BC break that is affecting real apps.
The biggest issue actually is that the thrown object is of type `Error`. That forces catching any error and skip any of them when calling `serialize()`. That's calling for debugging nightmares - no code should silence `Error`. Right now, we have to, see e.g. https://github.com/symfony/symfony/pull/36336, which I'd like to *not* merge.

Reply:
Agree with @TysonAndre that it's best to preserve the exception handling code here. It's not really important (worst that will happen is some redundant notice/warning), but if it's already there we can as well keep it :)

Reply:
The exception handling code is back.